# Binding partner prediction for DYNLL1/2 proteins

This repository contains scripts to identify binding motifs for DYNLL1/2 proteins in the human proteome based on a set of known binding sites. Only intracellular disordered regionson of proteins are considered.  

## Requirements:

* [iupred](https://iupred2a.elte.hu/) to annotate disordered segments in proteins.

## Usage:

#### 1. Converting known binding sites to scores

Reading a list of binding sites and calculating the position specific distribution of the occurring aminoacids. (all peptied must be aligned) The output of the program is the "scoring matrix", which is used for scoring the putative binding sites

`src/motif2score.pl -i <infile> -o <outfile>`

#### 2. Pre-processing input data

Protein sequences are annotated based on cellular localization and disorderedness: all extracellular, transmembrane and ordered regions are masked out and will not be included in the prediction. Sequence topology is pulled from [Uniprot API](https://www.uniprot.org/help/programmatic_access), disorderedness is predicted by iupred. 

`src/seq_cleaner.pl -i <inputfile.fasta> -o <outputfile>` 

* *inputfile.fasta* - protein sequences downloaded from [UniProt](https://www.uniprot.org/)

#### 3. Predicting binding sites

Uses the output of the `seq_cleaner.pl` and `motif2score.pl`. Writes two kinds of output: a score distribution and a detailed output with the sequences, scores and the positions etc. 

`src/predictor.pl -s <scorefile> -i <inputfile> -o <outputfile> -d <score_distrib>`

* *scorefile*: file generated by `motif2score.pl`
* *inputfile*: file generated by `seq_cleaner.pl`

#### 4. Generating random scores

To estimate a score threshold for true binging partners, the scoring matrix is shuffled 1000 times and the predictin is repeated with these shuffled scores. 

`src/score_scrabler.pl -s <scorefile>`

* *scorefile*: file generated by `motif2score.pl`

## Reference:

For more information on the method, see:

Rapali, P. et al. “Directed evolution reveals the binding motif preference of the LC8/DYNLL hub protein and predicts large numbers of novel binders in the human proteome.” [link](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0018818)

## License:

MIT License.

